cdap-tms/src/main/java/io/cdap/cdap/messaging/store/MessageTableKey.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.cdap.messaging.store;
+
+import io.cdap.cdap.api.common.Bytes;
+
+/**
+ * Row key used for an entry in a MessageTable.
+ */
+public class MessageTableKey {
+  private final byte[] rowKey;
+  private final int topicLength;
+  private long publishTimestamp;
+  private short sequenceId;
+
+  public MessageTableKey(byte[] topic) {
```

Comment:
I would suggest having this as private constructor and have two static methods, `fromTopic` and `fromBytes` to construct an instance of this. Usually it is not good to have a mix of static factory method and constructor together.